spec/kitchen_spec/directions/bake_link_placeholders_spec.rb

```diff
@@ -98,9 +108,15 @@
     end
   end
 
-  it 'logs a warning' do
-    allow($stdout).to receive(:puts)
-    expect($stdout).to receive(:puts).with("warning! could not find a replacement for '[link]' on an element with ID 'invalid_key'")
+  it 'logs a warning when href is invalid' do
+    expect(Warning).to receive(:warn).with("warning! could not find a replacement for '[link]' on an element with ID 'invalid_key'\n")
     described_class.v1(book: invalid_book)
   end
+
+  it 'logs a warning when link has no href' do
+    expect(Warning).to receive(:warn).with(/warning! Link has no href on element: /)
+    expect(Warning).to receive(:warn).with(/warning! Link has no href on element: /)
```

Comment:
another way to say this:

```ruby
expect(Warning).to receive(:warn).with(/blah/).twice
```

[more on expect counts](https://relishapp.com/rspec/rspec-mocks/docs/setting-constraints/receive-counts)